src/planner/ompl/Planner.cpp

```diff
@@ -317,6 +317,68 @@ trajectory::InterpolatedPtr planCRRTConnect(
                   std::move(_interpolator), _maxPlanTime);
 }
 
+//=============================================================================
+trajectory::InterpolatedPtr simplifyOMPL(statespace::StateSpacePtr _stateSpace,
+                                           statespace::InterpolatorPtr _interpolator,
+                                           distance::DistanceMetricPtr _dmetric,
+                                           constraint::SampleablePtr _sampler,
+                                           constraint::TestablePtr _validityConstraint,
+                                           constraint::TestablePtr _boundsConstraint,
+                                           constraint::ProjectablePtr _boundsProjector, 
+                                           double _maxDistanceBtwValidityChecks,
+                                           trajectory::InterpolatedPtr _originalTraj)
+{
+
+// Step 1: Generate the space information of OMPL type
+auto si = getSpaceInformation(
+      _stateSpace, _interpolator, std::move(_dmetric), std::move(_sampler),
+      std::move(_validityConstraint), std::move(_boundsConstraint),
+      std::move(_boundsProjector), _maxDistanceBtwValidityChecks);
+
+// Get the state space
+auto sspace = ompl_static_pointer_cast<GeometricStateSpace>(
+      si->getStateSpace());
+
+
+// Step 2: Convert the AIKIDO trajectory to Geometric Path 
+::ompl::geometric::PathGeometric *path = new ::ompl::geometric::PathGeometric(si);
+
+for (size_t idx = 0; idx < _originalTraj->getNumWaypoints(); ++idx) 
+{
+      auto ompl_state = sspace->allocState(_originalTraj->getWaypoint(idx));
+      path->append(ompl_state);
+      sspace->freeState(ompl_state);
+}
+
+// Step 3: Use the OMPL methods to simplify the path
+::ompl::geometric::PathSimplifierPtr simplifier;
+simplifier.reset(new ::ompl::geometric::PathSimplifier(si));
+bool const shortened = simplifier->shortcutPath(*path, 1, 1, 1.0, 0.005); 
+
+// Step 4: Convert the simplified geomteric path to AIKIDO untimed trajectory
+if(shortened)
+{
+  auto returnTraj = std::make_shared<trajectory::Interpolated>(
+        std::move(_stateSpace), std::move(_interpolator));
+
+  for (size_t idx = 0; idx < path->getStateCount(); ++idx) 
+  {
+        const auto *st =
+            static_cast<GeometricStateSpace::StateType *>(
+                path->getState(idx));
+        // Arbitrary timing
+        returnTraj->addWaypoint(idx, st->mState);
+  }
+  return returnTraj;  
 }
+else
+{
+  return _originalTraj;
```

Comment:
Returning `_originalPath` here is dangerous because it aliases the input argument. This could very easily cause a bug because the user expects `_originalTraj` to be unmodified.

This is a bit of a moot point because:

1. It's not possible to return `_originalTraj` once we change the return value to an `unique_ptr`.
2. It's safe to return `returnTraj` even if `shortcutPath` returned `false`.